### ISSUE TITLE
crashing MFS with non-cached MerkleDAG

### ISSUE THREAD

**Issue by michel47** [ipfs/go-mfs#83]
Dear IPFS team,
I encountered a problem with some block were missing from a directory w/i the MFS tree,
in order to make the problem 'repeatable' I wrote the following script, every subsequent mfs command is timing out  ...

```sh
# adding a directory w/ missing part!

qmdir=$(ipfs add -n -r -Q .)
echo qmdir: $qmdir

sed -e "s/Qm1234/$qmdir/" > dir.json <<EOF
{ "data" : "CAE=",
  "links" : [
   { "Cid" : { "/" : "Qm1234" }, "Name" : "not-here" }
  ]
}
EOF

qm=$(cat dir.json | ipfs dag put -f protobuf)
echo qm: $qm;
ipfs files rm -r /dont-do-ls 2>/dev/null
ipfs files cp /ipfs/$qm /dont-do-ls
ipfs files stat /dont-do-ls

ipfs object get $qm --data-encoding base64
ipfs --timeout 2s ls /ipfs/$qm
ipfs files ls -l /
ipfs --timeout 2s files ls -l /dont-do-ls

ipfs --timeout 2s files stat /

true;
```

**Comment by schomatis** [comment]
Hi @michel47, running this locally doesn't time out for the current MFS version in `go-ipfs` v0.8.0, just returns `Error: merkledag: not found`. Could you confirm if this is still a problem or if we can close it?

**Comment by michel47** [comment]
The problem is still there, let me attached a transcript,
is you have ``merkledag: not found`` it might simply be that your ipfs daemon is not running.
I am also running v0.8.0

This is what I have executed :

```sh
ipfs get QmXBASi3m4yyuzMxuHhU1A3JpXmKjGZyKrtj2S1guJtbqQ -o issue83
cd issue83
sh test.sh
```

and this is the logfile from screen
```log
qmdir: QmSW8egCbo1JxHYB9oVAuwDVKrAVkWQMwbCu4sUWDqr6XN
qm: Qma2nqq9rYCk8dWv2zoroMKZRujDyccoEmKZ9xoAajRrcp
Qma2nqq9rYCk8dWv2zoroMKZRujDyccoEmKZ9xoAajRrcp
Size: 0
CumulativeSize: 54
ChildBlocks: 1
Type: directory
{"Links":[{"Name":"not-here","Hash":"QmSW8egCbo1JxHYB9oVAuwDVKrAVkWQMwbCu4sUWDqr6XN","Size":0}],"Data":"CAE="}
Error: context deadline exceeded
.../	QmY1iB7DKsDC6vdahHQ4gpcZrTCKLakpBoYnBB94srNxNP	0
dont-do-ls/	Qma2nqq9rYCk8dWv2zoroMKZRujDyccoEmKZ9xoAajRrcp	0
etc/	QmSrCo7j4JnfKJqyHXvAgiTZn1CZLe7FYAoG7yZgpS7DeT	0
my/	QmeD79eRi2QamUYumAf97sNMsgjRxaJkRWjW1p8DZesDU9	0
public/	QmUPtfEJBrfFEaBLXVx8NXBFWZwtZ9rRz45CsE8i5sMf2h	0
root/	Qmb5EWCGmiw6162BFCVqnd82cmQXFdgAsKMtbFfoZ8Y3fp	0
Error: context deadline exceeded
Error: context deadline exceeded

```

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
